azurelinuxagent/common/utils/restutil.py

```diff
@@ -170,6 +174,12 @@ def _get_http_proxy(secure=False):
     return host, port
 
 
+def cleanup_sas_tokens_from_urls(url):
+    if SAS_TOKEN_RETRIEVAL_REGEX.search(url):
+        return SAS_TOKEN_RETRIEVAL_REGEX.sub(r"\1" + REDACTED_TEXT + r"\3", url)
```

Comment:
There's no need to call re.search() and re.sub(). From the re.sub() documentation:

> Return the string obtained by replacing the leftmost non-overlapping occurrences of pattern in string by the replacement repl. If the pattern isn’t found, string is returned unchanged.

Delete the `if` line; just return what SAS_TOKEN_RETRIEVAL_REGEX.sub() returns.